### PR TITLE
Restrict the user till MFA setup is complete

### PR DESCRIPTION
**Pull request by cmjishnu**
Return an error when the user try to get/patch without completing MFA setup when MFA is enabled.

Tested by:

Enable MFA in the system.
Verified generate secret key message is returned when creating a redfish session
Verified generate secret key message is returned if get/patch is executed.

Fixes : https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=652534

### CODE REVIEW

File: include/dbus_privileges.hpp [hunk 1]
```diff
@@ -90,10 +94,22 @@ inline bool
         asyncResp->res.result(boost::beast::http::status::forbidden);
         if (req.session->isConfigureSelfOnly)
         {
-            redfish::messages::passwordChangeRequired(
-                asyncResp->res,
-                boost::urls::format("/redfish/v1/AccountService/Accounts/{}",
-                                    req.session->username));
+            if (req.session->isGenerateSecretKeyRequired)
+            {
```
Comment (raviteja-b) [inline include/dbus_privileges.hpp:98]: does it make sure order of throwing passwordChangeRequired first? when both passwordChangeRequired and isGenerateSecretKeyRequired set?

File: include/dbus_privileges.hpp [hunk 2]
```diff
@@ -49,10 +51,12 @@ inline bool
     session.userRole = userRole;
     BMCWEB_LOG_DEBUG("userName = {} userRole = {}", session.username, userRole);
 
+    session.isGenerateSecretKeyRequired = secretKeyRequired.value_or(false);
     // Set isConfigureSelfOnly based on D-Bus results.  This
     // ignores the results from both pamAuthenticateUser and the
     // value from any previous use of this session.
-    session.isConfigureSelfOnly = passwordExpired.value_or(false);
+    session.isConfigureSelfOnly = passwordExpired.value_or(false) ||
+                                  session.isGenerateSecretKeyRequired;
```
Comment (raviteja-b) [inline include/dbus_privileges.hpp:59]: passwordExpired.value_or(false) || secretKeyRequired.value_or(false)  ?

**Review by raviteja-b — COMMENTED** [review]
Please share test results

**Comment by raviteja-b** [comment]
please share upstream commit here

File: include/dbus_privileges.hpp (hunk 2 above)
Comment (abhilashraju) [reply include/dbus_privileges.hpp:59]: why are you using session.isGenerateSecretKeyRequired?

**Pull request closed without merging**